cmd/kube-apiserver/app/options/options.go

```diff
@@ -144,6 +146,11 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
 	s.EgressSelector.AddFlags(fss.FlagSet("egress selector"))
 	s.Admission.AddFlags(fss.FlagSet("admission"))
 
+	// TODO(RainbowMango): move it to genericoptions before next flag comes.
+	mfs := fss.FlagSet("metrics")
+	mfs.StringVar(&s.ShowHiddenMetricsForVersion, "show-hidden-metrics-for-version", s.ShowHiddenMetricsForVersion,
+		"The previous version(x.y) for which you want to show hidden metrics. Only the previous minor version is allowed.")
```

Comment:
What is this version? Which formats are accepted? What does the last sentence mean?

Reply:
Why not `--show-hidden-metrics` without any argument?

Reply:
This kind of flag is incompatible by design to the previous version on each version change.

Reply:
> Why not --show-hidden-metrics without any argument?

See the discussion above:
https://github.com/kubernetes/kubernetes/pull/84292#discussion_r338804880

Also updated to KEP:
[why not --show-hidden-metrics](https://github.com/kubernetes/enhancements/blob/73a9f147245bee11d6590e9c6e09c6ee349bcac7/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md#why-not-bool-flag)

Reply:
Ok, got that.

Another point: I have not read the KEP. I have no clue what this flag does. What does "show" mean? What does "hidden" mean? This must be super clear from the flag description.

Reply:
maybe `--show-deprecated-hidden-metrics" ?

Reply:
But "deprecated" means something else. "Hidden" is well-defined by the metrics stability KEP, where the lifecycle of metrics is defined as:

Alpha > Stable > Deprecated > Hidden

Reply:
Hidden is a state of a deprecated metric that is old enough no? At least that's how I always viewed it. This makes sense to me as well though.

Reply:
I'm also confused by this flag. Why didn't we stick with the original plan of allowing enabling a specific deprecated metric?

Reply:
That was the question I asked here on the KEP update: https://github.com/kubernetes/enhancements/pull/1358#discussion_r344640164

Reply:
> Why didn't we stick with the original plan of allowing enabling a specific deprecated metric?

Because we want to reserve the ability to disable a metric (regardless of whether it is stable or not) at application boot and that functionality is orthogonal to metrics being auto-disabled by metrics stability framework.

Reply:
The [original plan](https://github.com/kubernetes/enhancements/pull/946/files#diff-3d4ef632b2ce9fb3c84103dc166b833bR211) dictates providing a mechanism for turning off individual metrics via the command line. 

> ..all metrics should be able to be individually disabled by the cluster administrator, regardless of stability class. By default, all non-deprecated metrics will be automatically registered to the metrics endpoint unless explicitly blacklisted via a command line flag (i.e. '--disable-metrics=somebrokenmetric,anothermetric')

Reply:
I suggest: `The previous version for which you want to show hidden metrics. Only the previous minor version is meaningful, other values will be ignored. The format is <major>.<minor>, e.g.: 1.16`

Reply:
Given the discussion, I think it makes sense to add a comment here explaining the purpose of this (to avoid giving the admin a surprise at v N+2). This could also be said in the help text, e.g.: `The purpose of this format is make sure you have the opportunity to notice if the next release hides additional metrics, rather than being surprised when they are permanently removed in the release after that.`